### ISSUE TITLE
[1.20.1 - Mythicmobs [EVENTS] org.bukkit.event.player.PlayerBucketEmptyEvent

### ISSUE THREAD

**Issue by DieG0Dig** [IzzelAliz/Arclight#1088]
### I have confirmed that ...

- [X] Arclight is up to date
- [X] all dependencies are installed
- [X] all plugins and mods are up to date
- [X] unable to reproduce in Spigot
- [X] unable to reproduce in Forge

### Arclight version

arclight-1.20.1-1.0.1-SNAPSHOT-6562f32

### OS & Java versions

17

### Plugins and Mods

```raw
Plugins (17): LuckPerms, Chunky, PlaceholderAPI, AugmentedHardcore, Timer, ConditionalEvents, Vault, ServerVariables, SimpleScore, MythicMobs, Essentials, EssentialsChat, MythicLib, MMOItems, EssentialsSpawn, WolfyUtilities, CustomCrafting

Mods:
CustomPlayerModels-1.20-0.6.9a.jar, emotecraft-for-MC1.20.1-2.2.7-b.build.50-forge.jar, fancymenu_forge_2.14.9_MC_1.20.1.jar ,konkrete_forge_1.6.1-2_MC_1.20.jar, loadmyresources_forge_1.0.4_MC_1.19.3.jar, Pehkui-3.7.7+1.20.1-forge.jar, player-animation-lib-forge-1.0.2-rc1+1.20.jar
VideoPlayer-2.0.3-FORGE-1.20.X.jar
voicechat-forge-1.20.1-2.4.16.jar
```

### Description

When a player uses the bucket item, either with water, lava, snow, etc., it sends the following error on the console:

[02:31:41 ERROR] [MythicMobs]: [EVENTS] Exception thrown whilst handling event: org.bukkit.event.player.PlayerBucketFillEvent
[02:31:41 ERROR] [STDERR]: java.lang.NoSuchMethodError: 'org.bukkit.entity.Entity org.bukkit.entity.Player.getTargetEntity(int, boolean)'
[02:31:41 ERROR] [STDERR]:      at io.lumine.mythic.bukkit.adapters.BukkitSkillTriggers.onBucketFill(BukkitSkillTriggers.java:147)
[02:31:41 ERROR] [STDERR]:      at io.lumine.mythic.bukkit.utils.Delegates$ConsumerToBiConsumerSecond.accept(Delegates.java:144)
[02:31:41 ERROR] [STDERR]:      at io.lumine.mythic.bukkit.utils.events.functional.single.EventListener.execute(EventListener.java:131)
[02:31:41 ERROR] [STDERR]:      at TRANSFORMER/arclight@1.20.1-1.0.1-SNAPSHOT-6562f32/org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:70)
[02:31:41 ERROR] [STDERR]:      at TRANSFORMER/arclight@1.20.1-1.0.1-SNAPSHOT-6562f32/org.bukkit.plugin.SimplePluginManager.fireEvent(SimplePluginManager.java:589)
[02:31:41 ERROR] [STDERR]:      at TRANSFORMER/arclight@1.20.1-1.0.1-SNAPSHOT-6562f32/org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:576)
[02:31:41 ERROR] [STDERR]:      at TRANSFORMER/arclight@1.20.1-1.0.1-SNAPSHOT-6562f32/org.bukkit.craftbukkit.v1_20_R1.event.CraftEventFactory.getPlayerBucketEvent(CraftEventFactoryMixin.java:512)
[02:31:41 ERROR] [STDERR]:      at TRANSFORMER/arclight@1.20.1-1.0.1-SNAPSHOT-6562f32/org.bukkit.craftbukkit.v1_20_R1.event.CraftEventFactory.callPlayerBucketFillEvent(CraftEventFactoryMixin.java:488)
[02:31:41 ERROR] [STDERR]:      at TRANSFORMER/minecraft@1.20.1/net.minecraft.world.item.BucketItem.handler$bbi000$arclight$bucketFill(BucketItemMixin.java:548)
[02:31:41 ERROR] [STDERR]:      at TRANSFORMER/minecraft@1.20.1/net.minecraft.world.item.BucketItem.m_7203_(BucketItemMixin.java:69)
[02:31:41 ERROR] [STDERR]:      at TRANSFORMER/minecraft@1.20.1/net.minecraft.world.item.ItemStack.m_41682_(ItemStackMixin.java:275)
[02:31:41 ERROR] [STDERR]:      at TRANSFORMER/minecraft@1.20.1/net.minecraft.server.level.ServerPlayerGameMode.m_6261_(ServerPlayerGameModeMixin.java:286)
[02:31:41 ERROR] [STDERR]:      at TRANSFORMER/minecraft@1.20.1/net.minecraft.server.network.ServerGamePacketListenerImpl.m_5760_(ServerPlayNetHandlerMixin.java:2691)
[02:31:41 ERROR] [STDERR]:      at TRANSFORMER/minecraft@1.20.1/net.minecraft.network.protocol.game.ServerboundUseItemPacket.m_5797_(CPlayerTryUseItemPacketMixin.java:29)
[02:31:41 ERROR] [STDERR]:      at TRANSFORMER/minecraft@1.20.1/net.minecraft.network.protocol.game.ServerboundUseItemPacket.m_5797_(CPlayerTryUseItemPacketMixin.java:7)
[02:31:41 ERROR] [STDERR]:      at TRANSFORMER/minecraft@1.20.1/net.minecraft.network.protocol.PacketUtils.md64ebc5$lambda$ensureRunningOnSameThread$0$0(PacketThreadUtilMixin.java:537)
[02:31:41 ERROR] [STDERR]:      at TRANSFORMER/minecraft@1.20.1/net.minecraft.server.TickTask.run(TickTask.java:18)
[02:31:41 ERROR] [STDERR]:      at TRANSFORMER/minecraft@1.20.1/net.minecraft.util.thread.BlockableEventLoop.m_6367_(BlockableEventLoop.java:156)
[02:31:41 ERROR] [STDERR]:      at TRANSFORMER/minecraft@1.20.1/net.minecraft.util.thread.ReentrantBlockableEventLoop.m_6367_(ReentrantBlockableEventLoop.java:23)
[02:31:41 ERROR] [STDERR]:      at TRANSFORMER/minecraft@1.20.1/net.minecraft.server.MinecraftServer.m_6367_(MinecraftServerMixin.java:770)
[02:31:41 ERROR] [STDERR]:      at TRANSFORMER/minecraft@1.20.1/net.minecraft.server.MinecraftServer.m_6367_(MinecraftServerMixin.java:161)
[02:31:41 ERROR] [STDERR]:      at TRANSFORMER/minecraft@1.20.1/net.minecraft.util.thread.BlockableEventLoop.m_7245_(BlockableEventLoop.java:130)
[02:31:41 ERROR] [STDERR]:      at TRANSFORMER/minecraft@1.20.1/net.minecraft.server.MinecraftServer.m_129961_(MinecraftServerMixin.java:753)
[02:31:41 ERROR] [STDERR]:      at TRANSFORMER/minecraft@1.20.1/net.minecraft.server.MinecraftServer.m_7245_(MinecraftServerMixin.java:747)
[02:31:41 ERROR] [STDERR]:      at TRANSFORMER/minecraft@1.20.1/net.minecraft.util.thread.BlockableEventLoop.m_18699_(BlockableEventLoop.java:115)
[02:31:41 ERROR] [STDERR]:      at TRANSFORMER/minecraft@1.20.1/net.minecraft.server.MinecraftServer.m_130012_(MinecraftServerMixin.java:732)
[02:31:41 ERROR] [STDERR]:      at TRANSFORMER/minecraft@1.20.1/net.minecraft.server.MinecraftServer.m_130011_(MinecraftServerMixin.java:2074)
[02:31:41 ERROR] [STDERR]:      at TRANSFORMER/minecraft@1.20.1/net.minecraft.server.MinecraftServer.m_206580_(MinecraftServerMixin.java:251)
[02:31:41 ERROR] [STDERR]:      at java.base/java.lang.Thread.run(Thread.java:833)

If there are more than 10 players using the bucket the server crashes due to "Description: Accessing PalettedContainer from multiple threads" . also crashed with the following description: Description: Watching Server

### Step to reproduce

1.- The latest version of Mythicmobs MythicMobs 5.3.5-64893d49 by Lumine was used
2.- When the player interacts with buckets of water, the aforementioned error appears on the console
3.- If many players use any bucket the server will crash.

### Logs

https://paste.ubuntu.com/p/kQ8MYQsKSS/

**Comment by OneWorldStudio** [comment]
MysticMobs uses Paper events, and ArsLight only bukkit and spigot

**Comment by vgrynch** [comment]
Similar thing happens also with the modded content, and in the other ArcLight version - Horn (Minecraft 1.19.2)

@NeverPixel, it means this probably not related to the Paper vs Spigot/Bukkit.

In my case, When trying to use any non-vanilla bucket from any mod to place water, I'm getting the below exception:

```
Failed to handle packet net.minecraft.network.protocol.game.ServerboundUseItemPacket@5cfde6c9, suppressing error
```
and then
```
java.lang.NullPointerException: Cannot invoke "net.minecraft.world.item.ItemStack.m_41720_()" because "itemstack" is null
at org.bukkit.craftbukkit.v1_19_R1.event.CraftEventFactory.getPlayerBucketEvent(CraftEventFactoryMixin.java:443) ~[arclight-1.19.2-1.0.2-4a870b30.jar%23481!/:arclight-1.19.2-1.0.2-4a870b30]
at org.bukkit.craftbukkit.v1_19_R1.event.CraftEventFactory.callPlayerBucketEmptyEvent(CraftEventFactoryMixin.java:433) ~[arclight-1.19.2-1.0.2-4a870b30.jar%23481!/:arclight-1.19.2-1.0.2-4a870b30]
at net.minecraft.world.item.BucketItem.handler$eid000$arclight$bucketEmpty(BucketItemMixin.java:598) ~[server-1.19.2-20220805.130853-srg.jar%23805!/:?]
at net.minecraft.world.item.BucketItem.emptyContents(BucketItemMixin.java:135) ~[server-1.19.2-20220805.130853-srg.jar%23805!/:?]
```

If needed, I can help to narrow down the issue, but this seems to be quite an obvious one.

Mods that add non-vanilla buckets and that fail for me:
Blue Skies (Ventium Bucket - a lot of buckets with their custom lifeforms do not work!)
Windswept (wooden bucket - none work)

Strangely enough, the bucket from "Meadow" mod work...

Not critical for me, I can disable custom buckets, but Blue Skies custom ventium buckets with lifeforms - would be quite a loss to the game.

**Comment by IzzelAliz** [comment]
@vgrynch Give full logs.

**Comment by vgrynch** [comment]
(Sorry for late response)
I do not think it makes sense sending full logs in my case, it simply throws exception any time when use a bucket. Below is more complete one:

```
java.lang.NullPointerException: Cannot invoke "net.minecraft.world.item.ItemStack.m_41720_()" because "itemstack" is null
at org.bukkit.craftbukkit.v1_19_R1.event.CraftEventFactory.getPlayerBucketEvent(CraftEventFactoryMixin.java:443) ~[arclight-1.19.2-1.0.2-4a870b30.jar%23486!/:arclight-1.19.2-1.0.2-4a870b30]
at org.bukkit.craftbukkit.v1_19_R1.event.CraftEventFactory.callPlayerBucketEmptyEvent(CraftEventFactoryMixin.java:433) ~[arclight-1.19.2-1.0.2-4a870b30.jar%23486!/:arclight-1.19.2-1.0.2-4a870b30]
at net.minecraft.world.item.BucketItem.handler$eho000$arclight$bucketEmpty(BucketItemMixin.java:598) ~[server-1.19.2-20220805.130853-srg.jar%23815!/:?]
at net.minecraft.world.item.BucketItem.emptyContents(BucketItemMixin.java:135) ~[server-1.19.2-20220805.130853-srg.jar%23815!/:?]
at net.minecraft.world.item.BucketItem.m_142073_(BucketItemMixin.java:117) ~[server-1.19.2-20220805.130853-srg.jar%23815!/:?]
at com.legacy.blue_skies.items.buckets.VentiumBucketItem.m_7203_(VentiumBucketItem.java:98) ~[blue_skies-1.19.2-1.3.20.jar%23519!/:1.3.20]
at net.minecraft.world.item.ItemStack.m_41682_(ItemStackMixin.java:266) ~[server-1.19.2-20220805.130853-srg.jar%23815!/:?]
at net.minecraft.server.level.ServerPlayerGameMode.m_6261_(ServerPlayerGameModeMixin.java:286) ~[server-1.19.2-20220805.130853-srg.jar%23815!/:?]
at net.minecraft.server.network.ServerGamePacketListenerImpl.m_5760_(ServerPlayNetHandlerMixin.java:2712) ~[server-1.19.2-20220805.130853-srg.jar%23815!/:?]
at net.minecraft.network.protocol.game.ServerboundUseItemPacket.m_5797_(CPlayerTryUseItemPacketMixin.java:29) ~[server-1.19.2-20220805.130853-srg.jar%23815!/:?]
at net.minecraft.network.protocol.game.ServerboundUseItemPacket.m_5797_(CPlayerTryUseItemPacketMixin.java:7) ~[server-1.19.2-20220805.130853-srg.jar%23815!/:?]
at net.minecraft.network.protocol.PacketUtils.md738e10$lambda$ensureRunningOnSameThread$0$0(PacketThreadUtilMixin.java:537) ~[server-1.19.2-20220805.130853-srg.jar%23815!/:?]
at net.minecraft.server.TickTask.run(TickTask.java:18) ~[server-1.19.2-20220805.130853-srg.jar%23815!/:?]
at net.minecraft.util.thread.BlockableEventLoop.m_6367_(BlockableEventLoop.java:157) ~[server-1.19.2-20220805.130853-srg.jar%23815!/:?]
at net.minecraft.util.thread.ReentrantBlockableEventLoop.m_6367_(ReentrantBlockableEventLoop.java:23) ~[server-1.19.2-20220805.130853-srg.jar%23815!/:?]
at net.minecraft.server.MinecraftServer.m_6367_(MinecraftServerMixin.java:763) ~[server-1.19.2-20220805.130853-srg.jar%23815!/:?]
at net.minecraft.server.MinecraftServer.m_6367_(MinecraftServerMixin.java:157) ~[server-1.19.2-20220805.130853-srg.jar%23815!/:?]
at net.minecraft.util.thread.BlockableEventLoop.m_7245_(BlockableEventLoop.java:131) ~[server-1.19.2-20220805.130853-srg.jar%23815!/:?]
at net.minecraft.server.MinecraftServer.m_129961_(MinecraftServerMixin.java:746) ~[server-1.19.2-20220805.130853-srg.jar%23815!/:?]
at net.minecraft.server.MinecraftServer.m_7245_(MinecraftServerMixin.java:740) ~[server-1.19.2-20220805.130853-srg.jar%23815!/:?]
at net.minecraft.util.thread.BlockableEventLoop.m_18701_(BlockableEventLoop.java:140) ~[server-1.19.2-20220805.130853-srg.jar%23815!/:?]
at net.minecraft.server.MinecraftServer.m_130012_(MinecraftServerMixin.java:726) ~[server-1.19.2-20220805.130853-srg.jar%23815!/:?]
at net.minecraft.server.MinecraftServer.m_130011_(MinecraftServerMixin.java:2131) ~[server-1.19.2-20220805.130853-srg.jar%23815!/:?]
at net.minecraft.server.MinecraftServer.m_206580_(MinecraftServerMixin.java:244) ~[server-1.19.2-20220805.130853-srg.jar%23815!/:?]
at java.lang.Thread.run(Thread.java:833) [?:?]
```

**Comment by vgrynch** [comment]
Oh, I see you already made corrections, when will I be able to try it in 1.19.2? Or do I have to build it myself for the time being?

**Comment by vgrynch** [comment]
I have tried to apply a similar fix to 1.19.2 in a custom build - but it did not help. I still getting custom buckets crashing. Please, help to fix this (I can do my own build if needed, just need some guides where to look because I do not have much experience in Minecraft/Bukkit internals)

**Comment by chasiopen** [comment]
arclight 1.18.2 1.0.9 同样的问题，使用原版桶也会有这个问题，希望发布一个修复的版本。

**Comment by vgrynch** [comment]
Confirmed this is fixed in horn 1.0.4
